File: src/main/java/cn/don9cn/blog/action/system/rbac/SysPermissionAction.java
package cn.don9cn.blog.action.system.rbac;

import cn.don9cn.blog.action.BaseAction;
import cn.don9cn.blog.model.system.rbac.SysPermission;
import cn.don9cn.blog.plugins.daohelper.core.PageResult;
import cn.don9cn.blog.plugins.operaresult.core.OperaResult;
import cn.don9cn.blog.service.system.rbac.interf.SysPermissionService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * @Author: liuxindong
 * @Description: 权限action
 * @Create: 2017/10/16 14:27
 * @Modify:
 */
@RestController
@RequestMapping(value = "/system/rbac/permission")
public class SysPermissionAction extends BaseAction<SysPermission> {

    @Autowired
    private SysPermissionService sysPermissionService;

    @Override
    @PostMapping
    protected Object baseInsert(SysPermission sysPermission) {
        return sysPermissionService.baseInsert(sysPermission);
    }

    @Override
    protected Object baseInsertBatch(List<SysPermission> list) {
        return null;
    }

    @Override
    @PutMapping
    protected Object baseUpdate(SysPermission sysPermission) {
        return sysPermissionService.baseUpdate(sysPermission);
    }

    @Override
    protected Object baseRemove(String code) {
        return null;
    }

    @Override
    @DeleteMapping
    protected Object baseRemoveBatch(String codes) {
        return sysPermissionService.baseDeleteBatch(codes);
    }

    @Override
    @GetMapping
    protected Object baseFindById(String code) {
        return sysPermissionService.baseFindById(code);
    }

    @Override
    @GetMapping("/all")
    protected Object baseFindAll() {
        return sysPermissionService.baseFindAll();
    }

    @Override
    @GetMapping("/list")
    protected Object baseFindListByParams(SysPermission sysPermission) {
        return sysPermissionService.baseFindListByParams(sysPermission);
    }

    @Override
    @GetMapping("/page")
    protected Object baseFindByPage(int page, int limit, String startTime, String endTime, String orderBy, SysPermission sysPermission) {
        return sysPermissionService.baseFindByPage(new PageResult<>(page,limit,startTime,endTime,orderBy,sysPermission));
    }

    /**
     * 获取权限树
     * @return
     */
    @GetMapping("/tree")
    public Object getTree() {
        return sysPermissionService.getTree();
    }
}

File: src/main/java/cn/don9cn/blog/action/system/rbac/SysPermissionAction.kt
package cn.don9cn.blog.action.system.rbac

import cn.don9cn.blog.support.mongo.core.PageResult
import cn.don9cn.blog.action.BaseAction
import cn.don9cn.blog.model.system.rbac.SysPermission
import cn.don9cn.blog.service.system.rbac.SysPermissionService
import cn.don9cn.blog.support.action.ActionMsg
import cn.don9cn.blog.support.action.ActionMsgHandler
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["/system/rbac/permission"])
open class SysPermissionAction : BaseAction<SysPermission>() {

    @Autowired
    private var sysPermissionService: SysPermissionService? = null


    @PostMapping
    override fun baseInsert(t: SysPermission): ActionMsg {
        return ActionMsgHandler.insert(sysPermissionService!!.baseInsert(t))
    }

    override fun baseInsertBatch(list: List<SysPermission>): ActionMsg {
        return ActionMsgHandler.insert(sysPermissionService!!.baseInsertBatch(list))
    }

    @PutMapping
    override fun baseUpdate(t: SysPermission): ActionMsg {
        return ActionMsgHandler.update(sysPermissionService!!.baseUpdate(t))
    }

    override fun baseRemove(code: String): ActionMsg {
        return ActionMsgHandler.delete(sysPermissionService!!.baseDeleteById(code))
    }

    @DeleteMapping
    override fun baseRemoveBatch(codes: String): ActionMsg {
        return ActionMsgHandler.delete(sysPermissionService!!.baseDeleteBatch(codes))
    }

    @GetMapping
    override fun baseFindById(code: String): ActionMsg {
        return ActionMsgHandler.find(sysPermissionService!!.baseFindById(code))
    }

    @GetMapping("/all")
    override fun baseFindAll(): ActionMsg {
        return ActionMsgHandler.find(sysPermissionService!!.baseFindAll())
    }

    @GetMapping("/list")
    override fun baseFindListByParams(t: SysPermission): ActionMsg {
        return ActionMsgHandler.find(sysPermissionService!!.baseFindListByParams(t))
    }

    @GetMapping("/page")
    override fun baseFindByPage(page: Int, limit: Int, startTime: String?, endTime: String?, orderBy: String?, t: SysPermission): ActionMsg {
        return ActionMsgHandler.find(sysPermissionService!!.baseFindByPage(PageResult(page, limit, startTime, endTime, orderBy, t)))
    }

    @GetMapping("/tree")
    open fun getTree():ActionMsg{
        return ActionMsgHandler.find(sysPermissionService!!.getTree())
    }
}
